test/e2e/adminapi_cluster_getlogs.go

```diff
@@ -0,0 +1,120 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = Describe("[Admin API] Kubernetes get pod logs action", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+
+	const containerName = "e2e-test-container-name"
+	const podName = "e2e-test-pod-name"
+
+	When("in a standard openshift namespace", func() {
+		const namespace = "default"
+
+		It("should be able to get logs from a container of a pod", func() {
+			ctx := context.Background()
+			testGetPodLogsOK(ctx, containerName, podName, namespace)
+		})
+	})
+
+	When("in a customer namespace", func() {
+		const namespace = "e2e-test-namespace"
+
+		It("should be not be able to get logs from customer workload namespaces", func() {
+			ctx := context.Background()
+			testGetPodLogsFromCustomerNamespaceForbidden(ctx, containerName, podName, namespace)
```

Comment:
if you adhere to my previous review regarding checking the value of string `logs` we would want to receive an `error` here and validate that is has a value of `nil`.

Reply:
In this case a different status code would be seen than what was expected(Line 75) and that is where the slice trimming line would never get executed in my understanding.

> 	resp, err := adminRequest(ctx, http.MethodGet, "/admin"+resourceIDFromEnv()+"/kubernetespodlogs", params, nil, &logs)
	Expect(err).NotTo(HaveOccurred())
	Expect(resp.StatusCode).To(Equal(http.StatusOK))             <<<<<<<<<<<<